### PR TITLE
[2.x] Adds changes related to OnBehalfOf and Service Accounts feature

### PR DESCRIPTION
**Pull request by DarshitChanpura**
Associated PRs: https://github.com/opensearch-project/security/pull/3416

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by DarshitChanpura** [comment]
@RyanL1997 @scrawfor99 @cwperks Could you please review this?

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Implement Visitor Design pattern in QueryBuilder to enable the capability to traverse through the complex QueryBuilder tree. ([#10110](https://github.com/opensearch-project/OpenSearch/pull/10110))
 - Add capability to restrict async durability mode for remote indexes ([#10189](https://github.com/opensearch-project/OpenSearch/pull/10189))
 - Add Doc Status Counter for Indexing Engine ([#4562](https://github.com/opensearch-project/OpenSearch/issues/4562))
+- Adds changes related On-behalf-Of and Service Accounts feature ([#10258](https://github.com/opensearch-project/OpenSearch/pull/10258))
```
Comment (scrawfor99) [inline CHANGELOG.md:14]:
```suggestion
- Adds changes related to On-behalf-Of and Service Accounts feature ([#10258](https://github.com/opensearch-project/OpenSearch/pull/10258))
```

File: server/src/main/java/org/opensearch/discovery/InitializeExtensionSecurityRequest.java
Comment (scrawfor99) [inline server/src/main/java/org/opensearch/discovery/InitializeExtensionSecurityRequest.java:1]: I would recommend putting all logic inside the standard initialize transport action similar to what was done on the main branch.

File: server/src/main/java/org/opensearch/discovery/InitializeExtensionSecurityResponse.java
Comment (scrawfor99) [inline server/src/main/java/org/opensearch/discovery/InitializeExtensionSecurityResponse.java:1]: Same as above, I think we want to use a single transport action as was done in main.

File: server/src/main/java/org/opensearch/extensions/action/IssueServiceAccountResponse.java
Comment (scrawfor99) [inline server/src/main/java/org/opensearch/extensions/action/IssueServiceAccountResponse.java:1]: Same as above.

**Comment by DarshitChanpura** [comment]
@scrawfor99 @RyanL1997 please review this when you get a chance

**Comment by DarshitChanpura** [comment]
@scrawfor99 @RyanL1997 please let me know if this needs to be re-opened.

**Pull request closed without merging**